enarx-keep-sgx-shim/src/start.S

```diff
@@ -163,6 +197,40 @@ syscall:
 
     xor     %rcx,                   %rcx            # Clear %rcx
     zerof   %rcx                                    # Clear CPU flags
+
+    # Check for unexpected flags in xcr0 before
+    # executing xsavec
+    mov    $0,                      %rcx            # Specify xcr0 for xgetbv
+    xgetbv
+    and    $~0x3FF,                 %eax            # Mask xgetbv lowest 9 bits
+    cmp    $0,                      %eax            # No other flags should be set
+    jne    .Leexit                                  # Exit if extra flags set low
+    cmp    $0,                      %edx           # No other flags should be set
+    jne    .Leexit                                  # Exit if extra flags set hi
```

Comment:
Let's do this check in Rust in `entry()`.

Reply:
My thinking here is that the flags will be established at enclave creation and can't be changed. Therefore, we only need to check it once when the enclave first starts. This also means we can use the `exit(1)` syscall directly. It also implies we don't have to waste these instructions on every context switch.

Reply:
Makes sense.